src/analysis/openstreetmap/qgsosmdatabase.cpp

```diff
@@ -408,6 +408,11 @@ void QgsOSMDatabase::exportSpatiaLiteNodes( const QString& tableName, const QStr
     if ( t.count() == 0 )
       continue;
 
+    //check not null tags
+    for( int i = 0; i < notNullTagKeys.count(); ++i )
+      if ( !t.contains( notNullTagKeys[i] ) )
+        continue;
+
```

Comment:
This does not seem to have any effect. Probably you wanted to use the same approach with "skipNull" like in exportSpatiaLiteWays() ? It would be even better to put the logic into a separate utility function e.g. bool hasNullTagValues(const QgsOSMTags& t, const QStringList& tagKeys) - and use it in both export methods